test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -4225,9 +4225,10 @@ Describe 'Invoke-WebRequest and Invoke-RestMethod support Cancellation through C
         $pwsh = [PowerShell]::Create()
         $invoke = "`$result = $Command -Uri `"$Uri`" $Arguments"
         $task = $pwsh.AddScript($invoke).InvokeAsync()
-        Start-Sleep -Milliseconds $DelayMs
```

Comment:
The change doesn't honor the user specified timeout anymore. Is that an oversight?

Reply:
No, that's covered by the next line `Wait-UntilTrue` - it loops (up to 10 seconds) until the condition is true then returns either true or false depending on the result of the script block. The `|Should -BeTrue` ensures that if it times out, the test will fail, but it will wait longer to be sure.

Reply:
I know `Wait-UntilTrue` has its own default timeout, but the helper function `RunWithCancellation` here also takes a timeout value, which should be passed down to `Wait-UntilTrue`. Otherwise, the caller specified timeout value `$DelayMs` is no longer honored.

Reply:
However, if you pass `$DelayMs` down to `Wait-UntilTrue`, then I guess the test will be as flaky as today.
Maybe those tests should use a larger value for `-DelayMs`?

Reply:
there's another PR (#19532) to attempt to address this issue - I'm going to pull this from mine

Reply:
I agree the CTRL-C tests on IWR are inherently racy. They are setting up stalled download conditions where partial data has been received then cancelling the download with Stop to simulate CTRL-C (something that used to just hang). I had initially not added any CI tests because I was concerned they would be racy, but was asked to add them as part of the review process.

If you have any suggestions on how such a test could be better constructed to remove the raciness/slowness then I would be interested but I suspect it would require out-of-band signalling or time simulation which I don't think is practical in these types of tests.

Reply:
#19532 was merged. Mark this thread as resolved.